scripts/src/javatests/org/oppia/android/scripts/maven/RetrieveLicenseTextsTest.kt

```diff
@@ -0,0 +1,585 @@
+import org.mockito.kotlin.doReturn
+import org.mockito.kotlin.eq
+import org.mockito.kotlin.mock
+import org.oppia.android.scripts.proto.DirectLinkOnly
+import org.oppia.android.scripts.proto.ExtractedCopyLink
+import org.oppia.android.scripts.proto.License
+import org.oppia.android.scripts.proto.MavenDependency
+import org.oppia.android.scripts.proto.MavenDependencyList
+import org.oppia.android.scripts.proto.ScrapableLink
+import org.oppia.android.testing.assertThrows
+import org.w3c.dom.Element
+import org.w3c.dom.Node
+import org.xml.sax.InputSource
+import java.io.ByteArrayOutputStream
+import java.io.PrintStream
+import javax.xml.parsers.DocumentBuilderFactory
+
+/** Tests for [RetrieveLicenseTexts]. */
+class RetrieveLicenseTextsTest {
+  private val TOO_FEW_ARGS_FAILURE = "Too few arguments passed"
+  private val MAVEN_DEPENDENCY_LIST_NOT_UP_TO_DATE_FAILURE =
+    "maven_dependencies.textproto is not up-to-date"
+  private val SCRIPT_PASSED_INDICATOR = "Script execution completed successfully."
+
+  private val SCRAPABLE_LINK = "https://www.apache.org/licenses/LICENSE-2.0.txt"
+  private val DIRECT_LINK_ONLY = "https://developer.android.com/studio/terms.html"
+  private val EXTRACTED_COPY_ORIGINAL_LINK = "https://www.opensource.org/licenses/bsd-license"
+  private val EXTRACTED_COPY_LINK = "https://raw.githubusercontent.com/oppia/oppia-android-" +
+    "licenses/develop/simplified-bsd-license.txt"
+
+  private val mockLicenseFetcher by lazy { initializeLicenseFetcher() }
+
+  private val outContent: ByteArrayOutputStream = ByteArrayOutputStream()
+  private val originalOut: PrintStream = System.out
+
+  @Rule
+  @JvmField
+  var tempFolder = TemporaryFolder()
+
+  @Before
+  fun setUp() {
+    tempFolder.newFolder("values")
+    tempFolder.newFolder("scripts", "assets")
+    System.setOut(PrintStream(outContent))
+  }
+
+  @After
+  fun restoreStreams() {
+    System.setOut(originalOut)
+  }
+
+  @Test
+  fun testScript_noArguments_printsUsageStringAndThrowsException() {
+    val exception = assertThrows(Exception::class) {
+      RetrieveLicenseTexts(mockLicenseFetcher).main(arrayOf())
+    }
+
+    assertThat(exception).hasMessageThat().contains(TOO_FEW_ARGS_FAILURE)
+    assertThat(outContent.toString()).contains("Usage:")
+  }
+
+  @Test
+  fun testScript_oneArguments_printsUsageStringAndThrowsException() {
+    val exception = assertThrows(Exception::class) {
+      RetrieveLicenseTexts(mockLicenseFetcher).main(
+        arrayOf(
+          "${tempFolder.root}/values"
+        )
+      )
+    }
+
+    assertThat(exception).hasMessageThat().contains(TOO_FEW_ARGS_FAILURE)
+    assertThat(outContent.toString()).contains("Usage:")
+  }
+
+  @Test
+  fun testScript_dependencyListEmpty_failsWithNotUpToDateException() {
+    val mavenDependencyList = getMavenDependencyList()
+
+    val pbFile = tempFolder.newFile("scripts/assets/maven_dependencies.pb")
+    mavenDependencyList.writeTo(pbFile.outputStream())
```

Comment:
Does writeTo() close the stream? If not, this should probably be:

```suggestion
    pbFile.outputStream().use { mavenDependencyList.writeTo(it) }
```

To ensure that resources are properly closed. Ditto for other tests.

Reply:
Done.